src/TTextCodec.cpp

```diff
@@ -316,6 +382,51 @@ QString TTextCodec_869::convertToUnicode(const char *in, int length, ConverterSt
     return result;
 }
 
+// This one does have some invalid codes in the table so we need to also
+// need to spot any incoming (char)in values that correspond to the invalid
+// (replacement) codepoint:
+QString TTextCodec_medievia::convertToUnicode(const char *in, int length, ConverterState *state) const
+{
+    QString result;
+    bool headerDone = false;
+    int invalidCharacters = 0;
+    if (state) {
+        // zero is false:
+        if (!state->state_data[0]) {
```

Comment:
I think code is easier to understand like this - instead of an `if` that needs explanation, just be straightforward:

```suggestion
        if (state->state_data[0] == 0) {
```

Reply:
Does https://codebrowser.dev/qt5/qtbase/src/corelib/codecs/qtextcodec.cpp.html make the code easier to understand - not really I guess - and it isn't clear if this opaque value is used for anything other than a `boolean` value - so it feels cleaner to me to treat it as such and not introduce a numerical value if I don't have to...

Reply:
I'm not sure I am following - what this does is just simplify the code so you don't have to jump through a mental hoop

Reply:
:man_shrugging: Okay, I guess...